test/unit/scripts/test_pudl_null_cols.py

```diff
@@ -0,0 +1,322 @@
+    mock_pudl_paths = mocker.patch("pudl.scripts.pudl_null_cols.PudlPaths")
+    mock_pudl_paths.return_value = mock_paths
+
+    available_years = get_available_years(
+        table_name="test_table", date_column="report_date", max_year=2021
+    )
+    assert available_years == [2020, 2021]
+
+
+def test_infer_row_conditions(mocker):
+    """Test the infer_row_conditions function."""
+    # Mock the schema
+    mock_schema = mocker.MagicMock()
+    mock_schema.names = ["report_date", "column_a", "column_b", "column_c"]
+    mock_read_schema = mocker.patch("pudl.scripts.pudl_null_cols.pq.read_schema")
+    mock_read_schema.return_value = mock_schema
+
+    # Mock available years
+    mock_get_available_years = mocker.patch(
+        "pudl.scripts.pudl_null_cols.get_available_years"
+    )
+    mock_get_available_years.return_value = [2020, 2021, 2022]
+
+    # Mock null years for different columns
+    def mock_null_years_side_effect(table_name, column, **kwargs):
+        if column == "column_a":
+            return [2021]
+        if column == "column_b":
+            return [2020, 2022]
+        # column_c
+        return []
+
+    mock_get_null_years = mocker.patch("pudl.scripts.pudl_null_cols.get_null_years")
+    mock_get_null_years.side_effect = mock_null_years_side_effect
+
+    # Mock PudlPaths
+    mock_pudl_paths = mocker.patch("pudl.scripts.pudl_null_cols.PudlPaths")
+    mock_pudl_paths.return_value = mocker.MagicMock()
+
+    result = infer_row_conditions(table_name="test_table", date_column="report_date")
+
+    # Should only include columns with null years
+    assert "column_a" in result
+    assert "column_b" in result
+    assert "column_c" not in result
+
+    # Check the generated conditions
+    assert result["column_a"] == "EXTRACT(year FROM report_date) NOT IN (2021)"
+    assert result["column_b"] == "EXTRACT(year FROM report_date) = 2021"
+
+
+def test_main_invalid_table():
+    """Test CLI with invalid table name."""
+    runner = CliRunner()
+    result = runner.invoke(main, ["invalid_table"])
+
+    assert result.exit_code != 0
+    assert "Invalid table name" in result.output
+
+
+def test_main_no_table():
+    """Test CLI with no table specified."""
+    runner = CliRunner()
+    result = runner.invoke(main, [])
+
+    assert result.exit_code != 0  # Click will raise an error for missing argument
+
+
+def test_main_conflicting_options(mocker):
+    """Test CLI with conflicting --ignore-eia860m and --max-year options."""
+    # Mock ALL_TABLES to include our test table so we get past the table validation
+    mocker.patch("pudl.scripts.pudl_null_cols.ALL_TABLES", ["test_table"])
+
+    runner = CliRunner()
+    result = runner.invoke(
+        main, ["test_table", "--ignore-eia860m", "--max-year", "2021"]
+    )
+
+    assert result.exit_code != 0
+    assert "mutually exclusive" in result.output
+
+
+def test_main_yaml_output(mocker):
+    """Test that CLI outputs valid YAML."""
```

Comment:
```suggestion
    """Test that CLI converts row conditions to YAML syntax."""
```

(we're not running a validator here)